odoo/tools/xml_utils.py

```diff
@@ -114,11 +114,26 @@ def cleanup_xml_node(xml_node_or_string, remove_blank_text=True, remove_blank_no
     """
     xml_node = xml_node_or_string
 
+    def remove_control_characters(byte_node):
+        """
+        The characters to be escaped are the control characters #x0 to #x1F and #x7F (most of which cannot appear in XML)
+        [...] XML processors must accept any character in the range specified for Char:
+        `Char	   :: =   	#x9 | #xA | #xD | [#x20-#xD7FF] | [#xE000-#xFFFD] | [#x10000-#x10FFFF]`
+        source:https://www.w3.org/TR/xml/
+        """
+        return byte_node.translate(
+            None,
+            b'\x00\x01\x02\x03\x04\x05\x06\x07\x08'  # must accept \x09 | \x0A
+            b'\x0B\x0C'  # must accept \x0D
+            b'\x0E\x0F\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1A\x1B\x1C\x1D\x1E\x1F'
+            b'\x7F'
+        )
```

Comment:
Maybe this would be more readable by matching the docs? (not sure where your `\x7F` comes from)
```suggestion
        return re.sub(
            '[^'
            '\u0009'
            '\u000a'
            '\u000d'
            '\u0020-\ud7ff'
            '\ue000-\ufffd'
            '\U00010000-\U0010FFFF'
            ']'.encode(),
            b'',
            byte_node,
        )
```

Reply:
The `re.sub` was my first option but performance wise the `byte.translate` is much better
https://gist.github.com/yosa-odoo/1b8c1a84f39c1fe72d3c2db3cbd188ab#file-perf-37738081-txt-L45-L53

Reply:
> not sure where your \x7F comes from

"The characters to be escaped are the control characters #x0 to #x1F and __#x7F__ (most of which cannot appear in XML)"

Reply:
But do we really care about speed when building/parsing the xml + i/o from the database is going to be much more than this?
I'd prefer having something more complete and a direct match to the docs.